app/src/main/java/org/simple/clinic/user/ReportPendingRecordsToAnalytics.kt

```diff
@@ -0,0 +1,78 @@
+package org.simple.clinic.user
+
+import io.reactivex.Completable
+import io.reactivex.rxkotlin.Singles
+import org.simple.clinic.analytics.Analytics
+import org.simple.clinic.bp.BloodPressureMeasurement
+import org.simple.clinic.bp.BloodPressureRepository
+import org.simple.clinic.drugs.PrescribedDrug
+import org.simple.clinic.drugs.PrescriptionRepository
+import org.simple.clinic.medicalhistory.MedicalHistory
+import org.simple.clinic.medicalhistory.MedicalHistoryRepository
+import org.simple.clinic.overdue.Appointment
+import org.simple.clinic.overdue.AppointmentRepository
+import org.simple.clinic.patient.PatientProfile
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.patient.SyncStatus.PENDING
+import org.threeten.bp.Instant
+import javax.inject.Inject
+
+class ReportPendingRecordsToAnalytics @Inject constructor(
+    private val patientRepository: PatientRepository,
+    private val bloodPressureRepository: BloodPressureRepository,
+    private val appointmentRepository: AppointmentRepository,
+    private val prescriptionRepository: PrescriptionRepository,
+    private val medicalHistoryRepository: MedicalHistoryRepository
+) {
+  fun report(): Completable {
+    return Singles
+        .zip(
+            patientRepository.recordsWithSyncStatus(PENDING),
```

Comment:
As discussed personally with @vinaysshenoy, we can use `SynceableRepository#pendingSyncRecordsCount`.

Reply:
I just realised that we can't do this. As part of pushing the records, we also push the oldest `updatedAt` timestamp to Analytics which will be used to figure out since when records have been lost. We'll need to load all the records anyway to compare the timestamps.